src/codegen/string_pool.zig

```diff
@@ -0,0 +1,37 @@
+const std = @import("std");
+
+const Allocator = std.mem.Allocator;
+pub const StringPool = struct {
```

Comment:
To store intermediate strings like names with transformed cases or variable names. It's otherwise difficult to manage their lifetimes.

Reply:
Yeah I saw that. I made similar helpers for casing.

The string pool is because it's very convenient to store intermediate strings and leads to more readable `writer.print("const {s} = struct {{");`s. It's also necessary to store them somewhere to handle identifier collisions. A BufferedWriter is another more memory-efficient choice for that, but then you still have to store the views into its buffer somewhere.